### ISSUE TITLE
false positive (extend_from_slice)

### ISSUE THREAD

**Issue by wagnerf42** [Manishearth/rust-clippy#1258]
hi, I have two kinds of vectors.
one is real stuff (type T) and the other one is references
on the real stuff.

so if v1 is real stuff and v2 references and I want
to extend v2 with the references on the stuff in v1 I do:
v2.extend(v1.iter())

I then get : use of `extend` to extend a Vec by a slice, #[warn(extend_from_slice)]

I don't see how to extend from slice in this case
[tree.txt](https://github.com/Manishearth/rust-clippy/files/505848/tree.txt)

see included example file

**Comment by wagnerf42** [comment]
I'm no rust expert and I might be wrong though.

**Comment by bluss** [comment]
You are right about that.

**Comment by Arnavion** [comment]
Smaller repro:

```rs
fn main() {
    struct S;

    let foo: Vec<S> = vec![];
    let foo = &foo; // Assume foo must be an &Vec<S> for some reasons

    let mut bar: Vec<&S> = vec![];
    bar.extend(foo); // IntoIterator for &Vec<T> produces Iterator<Item = &T>, so this works.
}
```

There's no easy way I can see to go from `&Vec<T>` to `&[&T]` that `extend_from_slice` would require.

**Comment by bluss** [comment]
The extend from slice lint is outdated anyway due to specializations in upstream rust.
